manifest/docker_schema2.go

```diff
@@ -279,10 +293,26 @@ func (m *Schema2) Inspect(configGetter func(types.BlobInfo) ([]byte, error)) (*t
 		Variant:       s2.Variant,
 		Os:            s2.OS,
 		Layers:        layerInfosToStrings(m.LayerInfos()),
+		LayersDetail:  imgInspectLayersFromLayerInfos(m.LayerInfos()),
+		History:       s2.History,
+		Author:        s2.Author,
+		Size:          s2.Size,
 	}
 	if s2.Config != nil {
 		i.Labels = s2.Config.Labels
 		i.Env = s2.Config.Env
+		i.Config.Env = s2.Config.Env
+		i.Config.Labels = s2.Config.Labels
+		i.Config.User = s2.Config.User
+		i.Config.Volumes = s2.Config.Volumes
+		i.Config.Entrypoint = s2.Config.Entrypoint
+		for key, value := range s2.Config.ExposedPorts {
+			exposedPorts := make(map[string]struct{})
+			exposedPorts[string(key)] = value
+			i.Config.ExposedPorts = exposedPorts
+		}
+		i.Config.StopSignal = s2.Config.StopSignal
+		i.Config.WorkingDir = s2.Config.WorkingDir
```

Comment:
See the `OCIConfig` comment… this feels like a code that shouldn’t need to exist.